clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/security/EcsCredentialsLifeCycleHandler.java

```diff
@@ -0,0 +1,160 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.cats.agent.Agent;
+import com.netflix.spinnaker.cats.module.CatsModule;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.provider.EcsProvider;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.ContainerInstanceCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.EcsCloudMetricAlarmCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.EcsClusterCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.IamPolicyReader;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.IamRoleCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.ScalableTargetsCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.SecretCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.ServiceCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.ServiceDiscoveryCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.TargetHealthCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.TaskCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.TaskDefinitionCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.agent.TaskHealthCachingAgent;
+import com.netflix.spinnaker.clouddriver.ecs.provider.view.EcsAccountMapper;
+import com.netflix.spinnaker.clouddriver.security.ProviderUtils;
+import com.netflix.spinnaker.credentials.CredentialsLifecycleHandler;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+@Lazy
+public class EcsCredentialsLifeCycleHandler
+    implements CredentialsLifecycleHandler<NetflixECSCredentials> {
+  protected final EcsProvider ecsProvider;
+  protected final AmazonClientProvider amazonClientProvider;
+  protected final AWSCredentialsProvider awsCredentialsProvider;
+  protected final Registry registry;
+  protected final IamPolicyReader iamPolicyReader;
+  protected final ObjectMapper objectMapper;
+  protected final CatsModule catsModule;
+  protected final EcsAccountMapper ecsAccountMapper;
+
+  @Override
+  public void credentialsAdded(@NotNull NetflixECSCredentials credentials) {
+    // To be implemented once CredentialsRepository is implemented.
+    //    if (credentials instanceof NetflixAssumeRoleEcsCredentials) {
+    //      ecsAccountMapper.addaddMapEntry(((NetflixAssumeRoleEcsCredentials)
```

Comment:
nit: I know this is currently commented out, but: `addaddMapEntry` --> `addMapEntry`